week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/cart/Cart.ts

```diff
@@ -0,0 +1,66 @@
+import {IProduct, IProductWithPrice, Product, ProductWithPrice} from '../product';
+
+class Cart<ItemType extends Product> {
+  private _items: ItemType[] = [];
+
+  public get items(): ItemType[] {
+    return this._items;
+  }
+
+  public getItem(id: string): ItemType | null {
+    return this._items.find(item => item.id == id) || null;
+  }
+
+  public addItem(item: ItemType): void {
+    const previousItem = this.getItem(item.id);
+
+    if (previousItem) {
+      previousItem.amount += item.amount;
+
+      return;
+    }
+
+    this.items.push(item);
+  }
+
+  public removeItem(id: string): void {
+    const previousItem = this.getItem(id);
+
+    if (!previousItem) {
+      return console.log(`Warning: item with given ID not exists in cart`);
```

Comment:
sądzę że tutaj powinieneś iść konsekwentnie.
Skoro Twój model danych @DariuszGala pilnuje np. `amount` i rzuca wyjątek jak jest niewłaściwie
to tutaj też bym robił `throw` zamiast `console.log`

i wyżej w `main.ts` był gotowy z `try...catch`